[Python] querying articles batch 2 [Replace C# samples] (#1798)
ravendb/docs: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/how-to-project-query-results.python.markdown

* Only a single projection request can be made per query.  
  Learn more in [single projection per query](../../../client-api/session/querying/how-to-project-query-results#single-projection-per-query).

### The cost of projections:

* Queries in RavenDB do not allow any computation to occur during the query phase.  
  However, you can perform any [calculations](../../../client-api/session/querying/how-to-project-query-results#example---projection-with-calculations) 
* inside the projection.

* But while calculations within a projection are allowed, having a very complex logic can impact query performance.  
  So RavenDB limits the total time it will spend processing a query and its projections.  
  Exceeding this time limit will fail the query. This is configurable, see the following configuration keys:  
      * [Databases.QueryTimeoutInSec](../../../server/configuration/database-configuration#databases.querytimeoutinsec)
      * [Databases.QueryOperationTimeoutInSec](../../../server/configuration/database-configuration#databases.queryoperationtimeoutinsec)

{PANEL/}

## Projection Methods

{PANEL: select_fields_query_data}

> * The most common way to perform a query with a projection is to use the `select_fields_query_data` method.  

Reviewer: ```suggestion
* The most common way to perform a query with a projection is to use the `select_fields` or `select_fields_query_data` methods.  
```

Author: done

* You can specify what fields from the document you want to retrieve and even provide a complex expression.

### Example - Projecting individual fields of the document:

{CODE-TABS}
{CODE-TAB:python:Query projections_1@ClientApi\Session\Querying\HowToProjectQueryResults.py /}
{CODE-TAB-BLOCK:sql:RQL}
from "Companies"

> select Name, Address.City as City, Address.Country as Country

Reviewer: ```suggestion
select name, address.city as City, address.country as Country
```

Author: done

{CODE-TAB-BLOCK/}
{CODE-TABS/}

### Example - Projecting arrays and objects:

{CODE-TABS}
{CODE-TAB:python:Query projections_2@ClientApi\Session\Querying\HowToProjectQueryResults.py /}
{CODE-TAB-BLOCK:sql:RQL}
// Using simple expression:
from "Orders"

> select ShipTo, Lines[].ProductName as ProductNames

Reviewer: ```suggestion
select ship_to, lines[].product_name as product_names
```

Author: done

// Using JavaScript object literal syntax:
from "Orders" as x
select {

>     ShipTo: x.ShipTo, 

Reviewer: lowercase the variable names in these examples
we should assume that the data that we want to fetch now using projections were stored using the Python client, so the field names would be snake_cased :]

Reviewer: but we should `snake_case` here **only** the variable names of the entities

Author: As agreed, waiting with this because the field names are planned to match those of sample data docs
